src/linux_mcp_server/tools/system_info.py

```diff
@@ -350,156 +511,272 @@ async def get_memory_information(
-                host=host,
-            )
-
-            if returncode == 0:
-                info.append("=== CPU Architecture (lscpu) ===")
-                info.append(stdout)
-        except FileNotFoundError:
-            info.append("CPU info: lscpu command not available")
-
-        # Try lspci for PCI devices
-        try:
-            returncode, stdout, stderr = await execute_command(
-                ["lspci"],
-                host=host,
-            )
-
-            if returncode == 0:
-                pci_lines = stdout.strip().split("\n")
-
-                info.append("\n=== PCI Devices ===")
-                # Show first 50 devices to avoid overwhelming output
-                for line in pci_lines[:50]:
-                    info.append(line)
-
-                if len(pci_lines) > 50:
-                    info.append(f"\n... and {len(pci_lines) - 50} more PCI devices")
-        except FileNotFoundError:
-            info.append("\nPCI devices: lspci command not available")
-
-        # Try lsusb for USB devices
-        try:
-            returncode, stdout, stderr = await execute_command(
-                ["lsusb"],
-                host=host,
-            )
-
-            if returncode == 0:
-                info.append("\n\n=== USB Devices ===")
-                info.append(stdout)
-        except FileNotFoundError:
-            info.append("\nUSB devices: lsusb command not available")
-
-        # Memory hardware info from dmidecode (requires root)
-        try:
-            returncode, stdout, stderr = await execute_command(
-                ["dmidecode", "-t", "memory"],
-                host=host,
-            )
-
-            if returncode == 0:
-                info.append("\n\n=== Memory Hardware (dmidecode) ===")
-                info.append(stdout)
-            elif "Permission denied" in stderr:
-                info.append("\n\nMemory hardware info: Requires root privileges (dmidecode)")
-        except FileNotFoundError:
-            info.append("\nMemory hardware info: dmidecode command not available")
-
-        if len(info) == 1:  # Only the header
-            info.append("No hardware information tools available.")
-
-        return "\n".join(info)
-    except Exception as e:
-        return f"Error getting hardware information: {str(e)}"
+
+    # Create a custom filter function that wraps filter_device_information
+    # to match the FilterFunc signature
+    async def custom_filter(parsed_data: ParsedData, fields: list[str] | None) -> ParsedData:
+        return await _filter_device_information(parsed_data, device_types=device_types, limit=limit)
+
+    # Use Rummager with custom collect, parse, and filter functions
+    rummager = DataPipeline(
+        collect_func=_collect_device_information,
+        parse_func=_parse_device_information,
+        filter_func=custom_filter,
+    )
+
+    # Commands list is empty since collection is dynamic
+    filtered_data = await rummager.process(commands=[], host=host)
```

Comment:
Passing an empty list seems a bit odd. If the caller doesn't have any commands to run, the API should not require commands.